src/hooks/core.rs

```diff
@@ -515,6 +481,16 @@ pub(crate) fn modify_damage(
         }
     }
 
+    // Check for PreventAllDamageAndEffects (Shinx's Hide)
+    if receiving_pokemon
+        .get_active_effects()
+        .iter()
+        .any(|effect| matches!(effect, CardEffect::PreventAllDamageAndEffects))
+    {
+        debug!("PreventAllDamageAndEffects: Preventing all damage and effects");
+        return 0;
```

Comment:
🔥

Reply:
I think we might be missing "effects" (i.e. putting a pokemon to sleep), but that sounds tricky! Maybe we can edit the `.add_effect` method, so that if a card has this CardEffect already, then the rest are noops.